### ISSUE TITLE
Missing sanity check on self.userDN

### ISSUE THREAD

**Issue by lecriste** [dmwm/WMCore#9410]
**Impact of the bug**
CRABClient

**Describe the bug**
`self.userDN` may be `None` [here](https://github.com/dmwm/WMCore/blob/ea825bf546699d3c9dd27d7ecbb5983e30e35787/src/python/WMCore/Credential/Proxy.py#L403
) and throw an exception:
https://hypernews.cern.ch/HyperNews/CMS/get/computing-tools/5305.html

**Expected behavior**
I expect to get an early error about the missing `self.userDN`

**Additional context and error message**
See HN link.

**Comment by amaltaro** [comment]
@lecriste having a look into this, we could simply cast that attribute to string and you wouldn't get any failures on those logging lines, however the real problem seems to actually be coming from this line in the CRABClient library:
https://github.com/dmwm/CRABClient/blob/master/src/python/CRABClient/CredentialInteractions.py#L272

So my suggestion is to check the output of that command before re-setting that `userDN` attr to whatever is the response of that call.

**Comment by lecriste** [comment]
Agree. Issue for CRAB then.
Still, I would protect the code from that error.

**Comment by belforte** [comment]
?? user DN can not be None.
Was this something like the user did not have a proper ~/.globus/usercert.pem ?

So are you Leonardo simply saying  "when accessing usercert.pem fails we want a more clear error then returning None" ?

IMHO this issue does a very poor job at descrbing what exactly the problem is, and where,
which is precondition to pointing to a solution

**Comment by lecriste** [comment]
> ?? user DN can not be None.
> Was this something like the user did not have a proper ~/.globus/usercert.pem ?
> 
> So are you Leonardo simply saying "when accessing usercert.pem fails we want a more clear error then returning None" ?

Yes. And if [getSubjectFromCert](https://github.com/dmwm/CRABClient/blob/7a83a98ae92e568b6c01db4d968109c15c0c231b/src/python/CRABClient/CredentialInteractions.py#L272) returns `None` we should not overwrite `userDN`.
 
> IMHO this issue does a very poor job at descrbing what exactly the problem is, and where,
> which is precondition to pointing to a solution
